From 363c084b54f0822e04b5a9e6622ea498e00aa897 Mon Sep 17 00:00:00 2001
From: David Kershner <dkershner@habana.ai>
Date: Fri, 13 Jun 2025 01:18:26 +0300
Subject: [PATCH] [SW-226950] sensord: add common state sensor type mask

The mask STATE_SENSOR_TYPE_MASK is being used in multiple locations.
Define once in the header and include the header.

Change-Id: Idd96e826ede2e90dbd7b5a0ff165fc3d6ff9b39e
Signed-off-by: David Kershner <dkershner@habana.ai>
---

diff --git a/arch/gaudi3/ipmi_sdr.c b/arch/gaudi3/ipmi_sdr.c
index 12cf470..d25849e 100644
--- a/arch/gaudi3/ipmi_sdr.c
+++ b/arch/gaudi3/ipmi_sdr.c
@@ -8,8 +8,6 @@
 #include <sensord.h>
 #include <ipmi_sdr.h>
 
-#define STATE_SENSOR_TYPE_MASK		BIT(PLDM_PDR_TYPE_STATE_SENSOR)
-
 #define IPMI_SDR_VERSION 0x51
 #define IPMI_DEVICE_SLAVE_ADDRESS 0xCE
 
diff --git a/include/sensord.h b/include/sensord.h
index dda6568..1ced43b 100644
--- a/include/sensord.h
+++ b/include/sensord.h
@@ -55,6 +55,7 @@
 
 #define NUMERIC_SENSOR_TYPE_MASK \
 	(BIT(PLDM_PDR_TYPE_NUMERIC_SENSOR) | BIT(PLDM_PDR_TYPE_COMPACT_NUMERIC_SENSOR))
+#define STATE_SENSOR_TYPE_MASK		BIT(PLDM_PDR_TYPE_STATE_SENSOR)
 
 struct sensord_sensor_entry;
 
diff --git a/src/pldm_pmc.c b/src/pldm_pmc.c
index 664e5ca..757eaa0 100644
--- a/src/pldm_pmc.c
+++ b/src/pldm_pmc.c
@@ -30,8 +30,6 @@
 
 #define PLDM_CRC8_POLYNOMIAL 0x07 /* x^8 + x^2 + x + 1 */
 
-#define STATE_SENSOR_TYPE_MASK		BIT(PLDM_PDR_TYPE_STATE_SENSOR)
-
 static struct pldm_pmc_state {
 	const struct pldm_pdr_record *records;
 	size_t num_records;
